driver-core/src/test/java/com/datastax/driver/core/HostConnectionPoolTest.java

```diff
@@ -1242,8 +1241,9 @@ public void should_wait_on_connection_if_zero_core_connections() throws Exceptio
 
             // Should create up to core connections.
             verify(factory, timeout(readTimeout).times(1)).open(any(HostConnectionPool.class));
-
+            Uninterruptibles.getUninterruptibly(request.connectionFuture, 5, TimeUnit.SECONDS);
```

Comment:
This line shouldn't be needed anymore since the completion of requestInitialized depends on connectionFuture.